internal/host/installcmd.go

```diff
@@ -227,3 +238,36 @@ func (i *installCmd) getDiskUnbootableCmd(ctx context.Context, host models.Host)
 	}
 	return formatCmds, nil
 }
+
+func constructHostInstallerArgs(cluster *common.Cluster, host *models.Host) (string, error) {
+	if cluster.ImageInfo.StaticIpsConfig == "" {
+		return host.InstallerArgs, nil
+	}
+
+	if host.InstallerArgs == "" {
+		newArgs := []string{"--copy-network"}
+		argsBytes, err := json.Marshal(newArgs)
+		if err != nil {
+			return "", err
+		}
+		return string(argsBytes), nil
+	}
+
+	var currentInstallerArgs []string
+	err := json.Unmarshal([]byte(host.InstallerArgs), &currentInstallerArgs)
+	if err != nil {
+		return "", err
+	}
+
+	// installer args already contain  command for network configuration
+	if funk.Contains(currentInstallerArgs, "--copy-network") {
+		return host.InstallerArgs, nil
+	}
+
+	currentInstallerArgs = append(currentInstallerArgs, "--copy-network")
+	argsBytes, err := json.Marshal(currentInstallerArgs)
+	if err != nil {
+		return "", err
+	}
```

Comment:
IIUC, `--copy-network` may be set elsewhere. In that case setting it must be a function called from all flows that need it, IMO. "Re-packaging" installer agrs is not a good way to make sure it's set.

Reply:
--copy-network must be set in case we are using static ip. it might be also set by user via UpdateInstallerArgs for his own reasons. So we just verifying that copy-network is present in case we are setting static IPs, and that it is not present twice